src/ci/github-actions/jobs.yml

```diff
@@ -342,9 +342,10 @@ auto:
       RUSTC_RETRY_LINKER_ON_SEGFAULT: 1
       MACOSX_DEPLOYMENT_TARGET: 10.12
       SELECT_XCODE: /Applications/Xcode_15.2.app
-      NO_LLVM_ASSERTIONS: 1
-      NO_DEBUG_ASSERTIONS: 1
-      NO_OVERFLOW_CHECKS: 1
+      # XXX(jieyouxu): I want assertions
+      #NO_LLVM_ASSERTIONS: 1
+      #NO_DEBUG_ASSERTIONS: 1
+      #NO_OVERFLOW_CHECKS: 1
```

Comment:
I don't think this is enough, because you still have `DEPLOY` and not `DEPLOY_ALT` here:
https://github.com/rust-lang/rust/blob/4847d6a9d07d4be9ba3196f6ad444af2d7bdde72/src/ci/run.sh#L127-L130

... and that's set automatically by the image name here:
https://github.com/rust-lang/rust/blob/4847d6a9d07d4be9ba3196f6ad444af2d7bdde72/src/ci/scripts/setup-environment.sh#L30-L36

Reply:
I did a lil hack by changing the job name to `dist-x86_64-apple-alt`

Reply:
Also thank you for point this out, I would not have guessed otherwise